mpt_trie/src/builder.rs

```diff
@@ -48,9 +49,22 @@ impl<T: PartialTrie> PartialTrieBuilder<T> {
     /// Inserts variants of extension and leaf nodes into the builder.
     pub fn insert_short_node_variants_from_proof(&mut self, proof: Vec<Vec<u8>>) {
         for node in proof {
-            let bytes = rlp::decode_list::<Vec<u8>>(&node);
-            match bytes.len() {
-                2 => self.insert_short_node_variants(bytes),
+            let rlp = Rlp::new(&node);
+            match rlp.prototype().expect("rlp data is valid") {
+                Prototype::List(2) => {
+                    self.insert_short_node_variants(rlp.as_list().expect("valid list"))
+                }
+                Prototype::List(17) => {
```

Comment:
Just to confirm, the 17 is coming from the 16 children + the final value right?

Reply:
yes that is correct.